core/src/main/java/com/github/kaktushose/jda/commands/dispatching/middleware/impl/ConstraintMiddleware.java

```diff
@@ -49,11 +65,27 @@ public void accept(@NotNull InvocationContext<?> context) {
             var optionData = commandOptions.get(i);
             for (var constraint : optionData.constraints()) {
                 log.debug("Found constraint {} for parameter {}", constraint, optionData.type().getName());
-                if (!constraint.validator().apply(argument, constraint.annotation(), context)) {
-                    context.cancel(errorMessageFactory.getConstraintFailedMessage(context, constraint));
-                    log.debug("Constraint failed!");
-                    return;
-                }
+                Validator<Object, Annotation> validator = (Validator<Object, Annotation>) constraint.validator();
+
+                // TODO remove this, when proteus is upgraded
+                Proteus.global().register(Type.of(Member.class), Type.of(Member.class), Mapper.uni((member, _) -> MappingResult.lossless(member)));
+
+                Constraint constraintAnn = constraint.annotation().annotation(Constraint.class).orElseThrow();
```

Comment:
this should throw a more precise exception, at least mark as TODO for #197

Reply:
This should never occour, I think it's fine that way. But will give it a TODO

Reply:
if it never occurrs, then its fine